openedx/features/custom_fields/multiselect_with_other/db/fields.py

```diff
@@ -93,8 +93,9 @@ def to_python(self, value):
             if isinstance(value, list):
                 return value
             elif isinstance(value, string_type):
-                choices_str = value.replace(self.other_delimiter, '')
-                selected_choices = [choice for choice in choices_str.split(',') if choice.strip()]
+                choices_str = value.split(self.other_delimiter)
+                selected_choices = [choice for choice in choices_str[0].split(',') if choice.strip()]
+                selected_choices.append(choices_str[1])
```

Comment:
There are some small concerns like 
`choices_str = value.split(self.other_delimiter)` split returns a list but we are naming variable as it returns string. 
Use of indexes isn't much clear. 
How about making them more readable in following way ?
```
                _choices = value.split(self.other_delimiter)
                selected_choices_str = _choices[0]
                other_choice_str = _choices[1]
                selected_choices = [choice for choice in selected_choices_str.split(',') if choice.strip()]
                selected_choices.append(other_choice_str)
```

Reply:
We are splitting them based on pipe delimiter(|), It is defined in its own scope. Its not using anywhere else so I guess we don't have to define those options separately.

Reply:
@muak50 
yes it is defined in its own scope, But If you see the suggested code, it looks more readable in a sense that it is showing that string value is separated by other delimiter that is pipe sign. and variables are showing that first part of string is containing the selected choices string as variable named `selected_choices_str` and second or after that delimiter string contains `other_choice_str` other choice string ..